<?xml version="1.0"?>
<launch>
  <arg name="debug_"  					default="False" />  
  <arg name="blur_param_"			default="1" />
  <arg name="hsv_target_"			default="110" />
  <arg name="hsv_target2_"		default="100" />
  <arg name="hsv_threshold_"	default="10" />
  <arg name="contour_area_min_"	default="50" />
  <arg name="contour_area_max_"	default="500" />
  <arg name="contour_ratio_min_"	default="1" />
  <arg name="contour_ratio_max_"	default="15" />
  <arg name="aruco_detection_" 		default="False" />
  <arg name="image_topic_" 				default="/kinect2/sd/image_color_rect" />
  <arg name="cloud_topic_"				default="/kinect2/sd/points" />
	<arg name="desc_match_thresh_"	default="0.005" />
	<arg name="gc_size_"						default="0.005"/>
	<arg name="gc_threshold_"				default="3"/>
	<arg name="downsampling_size_"	default="0.01"/>
	<arg name="icp_max_iter_"				default="10" />
	<arg name="icp_corr_distance_"	default="0.003" />
	<arg name="plane_cut_"					default="0.01" />
	<arg name="sor_mean_"						default="50"/>
	<arg name="sor_thresh_"					default="1.0"/>
	<arg name="clus_tolerance_" 		default="0.005"/>
  <arg name="clus_size_min_"			default="3"/>
  <arg name="clus_size_max_"			default="100"/>
  <arg name="model_file_"					default="/projection_test/box2"/>
	
  <node name="kinect_projection" pkg="ivan_aruco" type="kinect_projection" output="screen"> 
		<param name="blur_param_"          	type="int"   	value="$(arg blur_param_)"/> 
		<param name="hsv_target_"          	type="int"   	value="$(arg hsv_target_)"/> 
		<param name="hsv_target2_"					type="int"		value="$(arg hsv_target2_)"/>
		<param name="hsv_threshold_"   			type="int"   	value="$(arg hsv_threshold_)"/> 
		<param name="contour_area_min_"    	type="int"   	value="$(arg contour_area_min_)"/> 
		<param name="contour_area_max_"     type="int"   	value="$(arg contour_area_max_)"/> 
		<param name="contour_ratio_min_"    type="double"   	value="$(arg contour_ratio_min_)"/> 
		<param name="contour_ratio_max_"    type="double"   	value="$(arg contour_ratio_max_)"/> 
		<param name="aruco_detection_"    	type="bool"   		value="$(arg aruco_detection_)"/> 
		<param name="image_topic_"					type="string"			value="$(arg image_topic_)"/>
		<param name="cloud_topic_"					type="string"			value="$(arg cloud_topic_)"/>
		<param name="gc_size_"							type="double"			value="$(arg gc_size_)"/>
		<param name="gc_threshold_"     		type="int"   			value="$(arg gc_threshold_)"/> 
		<param name="desc_match_thresh_"		type="double"			value="$(arg desc_match_thresh_)"/>
		<param name="debug_"								type="bool"				value="$(arg debug_)"/>
		<param name="downsampling_size_"		type="double"			value="$(arg downsampling_size_)"/>
		<param name="icp_max_iter_"					type="int"				value="$(arg icp_max_iter_)" />
		<param name="icp_corr_distance_"		type="double"			value="$(arg icp_corr_distance_)" />
		<param name="plane_cut_"						type="double"			value="$(arg plane_cut_)" />	
		<param name="sor_mean_"							type="int"				value="$(arg sor_mean_)"/>
		<param name="sor_thresh_"						type="double"			value="$(arg sor_thresh_)"/>
		<param name="clus_tolerance_"		type="double"			value="$(arg clus_tolerance_)"/>
		<param name="clus_size_min_"     type="int"   			value="$(arg clus_size_min_)"/> 
		<param name="clus_size_max_"     type="int"   			value="$(arg clus_size_max_)"/> 
		<param name="model_file_"				type="string"				value="$(arg model_file_)"/>
	</node>
	
</launch>


